emscripten-simulated-jerry-api/jerry.js

```diff
@@ -36,10 +36,13 @@
   };
 
   var hasInternalProps = function (value) {
-    return (canHaveInternalProps(value) && value.__jerryInternalProps);
+    return (canHaveInternalProps(value) && __jerry._jerryInternalPropsWeakMap.has(value));
```

Comment:
Need to remove `canHaveInternalProps`, because we the WeakMap's key can be any value. (Would fix the bug that only objects could be mapped back to `jerry_value_t`.)

Reply:
seems we can't do that, because the key of the weakmap must be "Object"
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WeakMap
> Keys of WeakMaps are of the type Object only. Primitive data types as keys are not allowed (e.g. a Symbol can't be a WeakMap key).

Reply:
Oh, I missed that, bummer :( But not a huge problem I think?

I guess this means that to find the `jerry_value_t` for non-Object values, we'll need to do a slow path (iterate through the `_jerryToHostValueMap`), whereas for Object values, we can use `_jerryInternalPropsWeakMap` as a fast way to find the `jerry_value_t`.

Reply:
good idea

Reply:
Oh, that exactly what we did now, because now we will keep the key-value pairs in the `_jerryToHostValueMap` forever